internal/infrastructure/kubernetes/deployment.go

```diff
@@ -27,13 +29,102 @@ const (
 	// envoyCfgVolMntDir is the directory name of the Envoy configuration volume.
 	envoyCfgVolMntDir = "config"
 	// envoyCfgFileName is the name of the Envoy configuration file.
-	envoyCfgFileName = "envoy.json"
+	envoyCfgFileName = "bootstrap.yaml"
 	// envoyHTTPPort is the container port number of Envoy's HTTP endpoint.
 	envoyHTTPPort = int32(8080)
 	// envoyHTTPSPort is the container port number of Envoy's HTTPS endpoint.
 	envoyHTTPSPort = int32(8443)
 )
 
+var bootstrapTmpl = template.Must(template.New(envoyCfgFileName).Parse(`
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: 19000
+dynamic_resources:
+  cds_config:
+    resource_api_version: V3
+    api_config_source:
+      api_type: GRPC
+      transport_api_version: V3
+      grpc_services:
+      - envoy_grpc:
+          cluster_name: xds_cluster
+      set_node_on_first_message_only: true
+  lds_config:
+    resource_api_version: V3
+    api_config_source:
+      api_type: GRPC
+      transport_api_version: V3
+      grpc_services:
+      - envoy_grpc:
+          cluster_name: xds_cluster
+      set_node_on_first_message_only: true
+node:
+  cluster: envoy-gateway-system
+  id: envoy-default
+static_resources:
+  clusters:
+  - connect_timeout: 1s
+    load_assignment:
+      cluster_name: xds_cluster
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: {{ .XdsServerAddress }}
+                port_value: 18000
+    http2_protocol_options: {}
+    name: xds_cluster
+    type: STRICT_DNS
+layered_runtime:
+  layers:
+    - name: runtime-0
+      rtds_layer:
+        rtds_config:
+          resource_api_version: V3
+          api_config_source:
+            transport_api_version: V3
+            api_type: GRPC
+            grpc_services:
+              envoy_grpc:
+                cluster_name: xds_cluster
+        name: runtime-0
+`))
+
+var (
+	// envoyGatewayService is the name of the Envoy Gateway service.
+	envoyGatewayService = "envoy-gateway"
+)
+
+// envoyBootstrap defines the envoy Bootstrap configuration.
+type bootstrapConfig struct {
+	// parameters defines configurable bootstrap configuration parameters.
+	parameters bootstrapParameters
+	// rendered is the rendered bootstrap configuration.
+	rendered string
+}
+
+// envoyBootstrap defines the envoy Bootstrap configuration.
+type bootstrapParameters struct {
+	// XdsServerAddress is the address of the XDS Server that Envoy is managed by.
+	XdsServerAddress string
+}
+
+// render the stringified bootstrap config in yaml format.
+func (b *bootstrapConfig) render() error {
+	buf := new(bytes.Buffer)
+	if err := bootstrapTmpl.Execute(buf, b.parameters); err != nil {
+		return fmt.Errorf("failed to render bootstrap config: %v", err)
+	}
+	b.rendered = buf.String()
```

Comment:
If we never need access it as a `[]byte` or mutate it (other than append), then it'd be more efficient to use a `strings.Builder` instead of a `bytes.Buffer`.

Reply:
https://github.com/envoyproxy/gateway/issues/225